library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java

```diff
@@ -480,11 +518,12 @@ public boolean isOnCache() {
     }
 
     public Uri getMyDownloadsUri() {
-        return ContentUris.withAppendedId(Downloads.Impl.CONTENT_URI, mId);
+        return mAllDownloadsUri;
```

Comment:
Access to `ContentUris` will fail at test. 

As well there is no need to calculate the value every time as it can be precomputed